File: backend/app/schemas/changePassword.py
from pydantic import BaseModel, field_validator, model_validator
import re

PASSWORD_REGEX = r"^(?=.*[A-Z]).{5,}$"

class ChangePasswordtRequest(BaseModel):
    password: str
    confirmPassword: str

    @field_validator("password", "confirmPassword")
    def validate_required(cls, v):
        if not v or not v.strip():
            raise ValueError("Este campo es obligatorio.")
        return v

    @field_validator("password")
    def validate_password_format(cls, v):
        if not re.match(PASSWORD_REGEX, v):
            raise ValueError(
                "La contraseña debe tener mínimo 5 caracteres y al menos una mayúscula."
            )
        return v

    @model_validator(mode="after")
    def passwords_match(self):
        if self.password != self.confirmPassword:
            raise ValueError("Las contraseñas no coinciden.")
        return self


class ChangePasswordResponse(BaseModel):
    data: object

File: frontend/public/js/changePassword.js
const passwordInput = document.getElementById("password");
const confirmPasswordInput = document.getElementById("confirmPassword");
const API_URL = window.APP_CONFIG.API_URL;

async function sendChangePassword() {
  const password = passwordInput.value.trim();
  const confirmPassword = confirmPasswordInput.value.trim();

  mensajeDiv.textContent = "";
  mensajeDiv.style.display = "none";


  if (!/^(?=.*[A-Z]).{5,}$/.test(password)) {
    mostrarError(
      "La contraseña debe tener mínimo 5 caracteres y al menos una mayúscula.",
    );
    return;
  }

  if (!password) {
    mostrarError("Por favor ingresa una contraseña.");
    return;
  }

  if (password !== confirmPassword) {
    mostrarError("Las contraseña no coinciden.");
    return;
  }

  try {
    const res = await fetch(`${API_URL}/api/change-password`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ password, confirmPassword }),
    });

    if (!res.ok) {
      const errText = await res.text();
      throw new Error(`HTTP ${res.status}: ${errText}`);
    }

    const data = await res.json(); // { answer: "..." }
    console.log("data", data);
  } catch (err) {
    mostrarError(err.message);
    return;
  }
}

function mostrarError(mensaje) {
  mensajeDiv.textContent = mensaje;
  mensajeDiv.style.display = "block";
}

sendBtn.addEventListener("click", sendLogin);

// Escuchar tecla Enter
userInput.addEventListener("keypress", (e) => {
  if (e.key === "Enter") {
    sendLogin();
  }
});

function togglePassword(inputId, element) {
    const input = document.getElementById(inputId);

    if (input.type === "password") {
        input.type = "text";
        element.classList.remove("fa-eye");
        element.classList.add("fa-eye-slash");
    } else {
        input.type = "password";
        element.classList.remove("fa-eye-slash");
        element.classList.add("fa-eye");
    }
}
